lib/devise_token_auth/url.rb

```diff
@@ -4,15 +4,10 @@ module DeviseTokenAuth::Url
 
   def self.generate(url, params = {})
     uri = URI(url)
+    query_params = Hash[URI.decode_www_form(uri.query || '')].merge(params)
```

Comment:
The problem is that `expiry` gets lost here in rails 4.2 (`/controllers/devise_token_auth/passwords_controller_test.rb:207`). I thought `URI` is from ruby, so it's weird